src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const onlyMen = century
+    ? people
+      .filter(men => men.sex === 'm' && Math.ceil(men.died / 100) === century)
+    : people.filter(men => men.sex === 'm');
+
+  const ages = onlyMen.map(year => year.died - year.born);
+
+  return ages.reduce((sum, age) => sum + age, 0) / ages.length;
```

Comment:
you repeated this logic 2 times, let's create separate function that will accept people array and will return average age value